src/auth/auth.ts

```diff
@@ -18,7 +18,12 @@ import {
 
 //#endregion
 
-const TOKENIZED_URL_REGEX = /1bb:\/\/([a-z]{3})-([a-z0-9]{5})(-[a-z]{4}[0-9]{2})?\/(.*)/;
+export const TOKENIZED_URL_REGEX = /1bb:\/\/([a-z]{3})-([a-z0-9]{5})(-[a-z]{4}[0-9]{2})?(:[0-9]+)?\/(.*)/;
+export const SCS_INDEX = 1;
+export const SERVICE_INDEX = 2;
+export const ZONE_INDEX = 3;
+export const LOCAL_PORT_INDEX = 4;
+export const ENDPOINT_INDEX = 5;
```

Comment:
now that i see how this is being consumed, i see why you need to export these. 

could we change the return type of `getUrl` in order to prevent these exports? if it returned an object that contained all the parts of the URL then the user of the method would already have it broken apart for them as opposed to having to break it apart themselves. 

```typescript

interface UrlParts {
  full_url: string;
  scs: string;
  service: string;
  zone: string;
  local_port: string | number;
  endpoint: string;
}

public static getUrl(tokenizedUrl: string, args?: BBAuthGetUrlArgs): Promise<UrlParts> {
  return {
    full_url: `https://${match[SCS_INDEX]}-${zone}.app.blackbaud.net/${match[SERVICE_INDEX]}/${match[ENDPOINT_INDEX]}`,
    scs: match[SCS_INDEX],
    service: match[SERVICE_INDEX]
    // etc etc
  }
}
```

or something along those lines. the consuming `getLocallyServedUrl` in blackbaud/skyux-http#50 won't need the regex since it'll already have the local port.

admittedly, this might be a breaking change, so maybe you want to do it in a _different_ method to not break the original.

Reply:
We wanted to avoid changing how the existing function works.  Which is why we have the export.  What I'd rather do is wrap the regexing into an entity which hides the whole pattern matching all together so the existing function doesn't know there is pattern match and the consumption in the `getLocallyServedUrl` doesn't know about it either.  This way if we move away from regex pattern matching the consuming code won't change.

You've confirmed what I thought we needed to do, so I might just go ahead and add that extra abstraction.

Reply:
yea, i can see a `getUrl` method maintain existing behavior. and a new `getUrlParts` (but hopefully with a better name) method that returns all the bits.

Reply:
addressed